node/cn/api_backend.go

```diff
@@ -112,6 +133,31 @@ func (b *CNAPIBackend) BlockByNumber(ctx context.Context, blockNr rpc.BlockNumbe
 	return block, nil
 }
 
+func (b *CNAPIBackend) BlockByHash(ctx context.Context, hash common.Hash) (*types.Block, error) {
+	block := b.cn.blockchain.GetBlockByHash(hash)
+	if block == nil {
+		return nil, fmt.Errorf("the block does not exist (block hash: %s)", hash.String())
+	}
+	return block, nil
+}
+
+func (b *CNAPIBackend) BlockByNumberOrHash(ctx context.Context, blockNrOrHash rpc.BlockNumberOrHash) (*types.Block, error) {
+	if blockNr, ok := blockNrOrHash.Number(); ok {
+		return b.BlockByNumber(ctx, blockNr)
+	}
+	if hash, ok := blockNrOrHash.Hash(); ok {
+		block, err := b.BlockByHash(ctx, hash)
+		if err != nil {
+			return nil, err
+		}
+		if block == nil {
+			return nil, fmt.Errorf("header found, but block body is missing")
+		}
```

Comment:
Unreachable case. `b.BlockByHash` doesn't return `(nil, nil)`.

Reply:
Oh,, I remove this code. Thank you!